src/firewall/functions.py

```diff
@@ -26,6 +33,41 @@
 }
 
 
+def str_to_bool(val, default=DEFAULT_SENTINEL, ignore_error=False):
+
+    is_default = False
+
+    if isinstance(val, str):
+        val2 = val.lower().strip()
+        if val2 in ("1", "y", "yes", "true", "on"):
+            return True
+        if val2 in ("0", "n", "no", "false", "off"):
+            return False
+        if val2 in ("", "default", "-1"):
+            is_default = True
+    elif val is None:
+        # None is (maybe) accepted as default value.
+        is_default = True
+    elif isinstance(val, bool):
+        # For convenience, also accept that the value is already a boolean.
+        return val
+
+    if is_default:
+        if default is DEFAULT_SENTINEL:
+            # No default is specified. We don't allow a default value.
+            pass
+        else:
+            return default
+
+    if ignore_error:
+        # We don't fail, we fallback to the default.
+        if default is DEFAULT_SENTINEL:
+            return False
+        return default
```

Comment:
it's not yet used, because all the places that reimplement parsing are not yet ported to `str_to_bool()` yet.

In general, it makes sense to opt-in to accept garbage input. The parameter is there, so that the function is useful in various use-cases, in particular, also in cases where we don't have a trusted input, but don't care to validate bogus data.

Reply:
The default arg is controlled by us in the code. We can trust it. Please remove this `ignore_error` argument.

Reply:
The behavior for these keys should be consistent. Either we accept "garbage" and use the defaults. Or we reject "garbage" and throw an error. It should not be key specific. Either throw a config error for them all or ignore the config error, throw a log, and use the defaults.

Reply:
this function is called `str_to_bool()`. It does exactly that, with two arguments:
- specifying a default value
- `ignore_error=True` to opt-in accepting garbage

Granted, there is additional the twist, that this doesn't only parse TRUE/FALSE, but also a ternary value `""`/`"-1"`/`default` to accept a third value. You get that mode with `str_to_bool(val, default=None)`. OK, you seem to not like that mode. You get the other two modes of strict-parsing (`default=<UNSET>`, `ignore_error=False`) or accept garbage (`default=<ANYTHING>`, `ignore_error=True`).

Reply:
note that these keys are not only parsed from file, but also set via D-Bus. On the current version of the branch, you can set a value to "default" or "" or "-1" (for boolean properties), and that will reset the default.

Strict input validation is important and useful to the user, to tell them when they do something wrong. Being annoying when they say "default" is not useful, it's clear what they want.

Reply:
I can change that, I don't really care. I find it kinda nice, to be graceful in the user input we accept. But it's fine.

The point is more than anything, that changing the behavior for all boolean properties (including what is accepted from D-Bus) can now be changed with
```diff
-        v = firewall.functions.str_to_bool(value, default=None)
+        v = firewall.functions.str_to_bool(value)
```

Will do that.